src/test/fuzz/addrman.cpp

```diff
@@ -0,0 +1,114 @@
+        }
+        case 4: {
+            (void)addr_man.GetAddr(fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, 4096), fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, 4096));
+            break;
+        }
+        case 5: {
+            const std::optional<CAddress> opt_address = ConsumeDeserializable<CAddress>(fuzzed_data_provider);
+            const std::optional<CNetAddr> opt_net_addr = ConsumeDeserializable<CNetAddr>(fuzzed_data_provider);
+            if (opt_address && opt_net_addr) {
+                addr_man.Add(*opt_address, *opt_net_addr, fuzzed_data_provider.ConsumeIntegralInRange<int64_t>(0, 100000000));
+            }
+            break;
+        }
+        case 6: {
+            std::vector<CAddress> addresses;
+            while (fuzzed_data_provider.ConsumeBool()) {
+                const std::optional<CAddress> opt_address = ConsumeDeserializable<CAddress>(fuzzed_data_provider);
+                if (!opt_address) {
+                    break;
+                }
+                addresses.push_back(*opt_address);
+            }
+            const std::optional<CNetAddr> opt_net_addr = ConsumeDeserializable<CNetAddr>(fuzzed_data_provider);
+            if (opt_net_addr) {
+                addr_man.Add(addresses, *opt_net_addr, fuzzed_data_provider.ConsumeIntegralInRange<int64_t>(0, 100000000));
+            }
+            break;
+        }
+        case 7: {
+            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
+            if (opt_service) {
+                addr_man.Good(*opt_service, fuzzed_data_provider.ConsumeBool(), ConsumeTime(fuzzed_data_provider));
+            }
+            break;
+        }
+        case 8: {
+            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
+            if (opt_service) {
+                addr_man.Attempt(*opt_service, fuzzed_data_provider.ConsumeBool(), ConsumeTime(fuzzed_data_provider));
+            }
+            break;
+        }
+        case 9: {
+            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
+            if (opt_service) {
+                addr_man.Connected(*opt_service, ConsumeTime(fuzzed_data_provider));
+            }
+            break;
+        }
+        case 10: {
+            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
+            if (opt_service) {
+                addr_man.SetServices(*opt_service, ServiceFlags{fuzzed_data_provider.ConsumeIntegral<uint64_t>()});
+            }
+            break;
+        }
+        case 11: {
+            (void)addr_man.Check();
+            break;
+        }
+        }
+    }
+    (void)addr_man.size();
+    if (fuzzed_data_provider.ConsumeBool()) {
+        while (fuzzed_data_provider.ConsumeBool()) {
+            addr_man.m_asmap.push_back(fuzzed_data_provider.ConsumeBool());
```

Comment:
Should this be at the beginning so that addrman GetGroup calls use a populated asmap?

Reply:
Thanks! Fixed :)